src/entry.rs

```diff
@@ -26,12 +28,15 @@ pub trait Entry: Debug + Eq + PartialEq {
     fn mtime(&self) -> UnixTime;
     fn size(&self) -> Option<u64>;
     fn symlink_target(&self) -> &Option<String>;
+    fn umode(&self) -> UnixMode;
+    fn owner(&self) -> Owner;
 
     /// True if the metadata supports an assumption the file contents have
     /// not changed.
     fn is_unchanged_from<O: Entry>(&self, basis_entry: &O) -> bool {
         basis_entry.kind() == self.kind()
             && basis_entry.mtime() == self.mtime()
             && basis_entry.size() == self.size()
+            && basis_entry.umode() == self.umode()
```

Comment:
Seems like this should also check the owner?

Reply:
Yes I think so